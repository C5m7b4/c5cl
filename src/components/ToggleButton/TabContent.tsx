import React from 'react';

export interface TabContentProps {
  id: string;
  activeTab: string;
  children: React.ReactNode;
}

const TabContent = (props: TabContentProps) => {
  const { id, activeTab, children } = props;

  return activeTab === id ? (
    <div className="tab-content">{children}</div>
  ) : null;
};

export default TabContent;
