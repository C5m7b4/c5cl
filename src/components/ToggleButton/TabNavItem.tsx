import React from 'react';

export interface TabNavItemProps {
  id: string;
  title: string;
  activeTab: string;
  setActiveTab: (s: string) => void;
}

const TabNavItem = (props: TabNavItemProps) => {
  const { id, title, activeTab, setActiveTab } = props;

  const handleClick = () => {
    setActiveTab(id);
  };

  return (
    <li
      onClick={handleClick}
      className={activeTab === id ? 'toggle-button-active-tab' : ''}
    >
      {title}
    </li>
  );
};

export default TabNavItem;
