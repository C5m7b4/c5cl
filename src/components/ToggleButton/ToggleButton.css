.toggle-button {
  width: 80%;
  height: auto;
  color: #e8f0f2;
  border-radius: 2rem;
  transition: height 500ms ease-in;
}

.toggle-button .outlet {
  padding: 10px;
  color: black;
}

.toggle-button ul.nav {
  width: 60%;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #39a2db;
  border-radius: 2rem;
  padding-left: 0px;
  background: #053742;
}

.toggle-button ul.nav li {
  width: 50%;
  padding: 1rem;
  text-align: center;
  cursor: pointer;
}

.toggle-button ul.nav li:first-child {
  border-radius: 0;
  border-bottom-left-radius: 2rem;
  border-top-left-radius: 2rem;
}

.toggle-button ul.nav li:last-child {
  border-radius: 0;
  border-bottom-right-radius: 2rem;
  border-top-right-radius: 2rem;
}

.toggle-button ul.nav li:hover {
  background: rgba(50, 224, 196, 0.15);
}

.toggle-button ul.nav li:active {
  background: #39a2db;
}
.toggle-button ul.nav li:hover:active {
  background: #39a2db;
}

.toggle-button-active-tab {
  background: #39a2db;
}

.toggle-button-active-tab:hover {
  background: #39a2db;
}

.first-tab p,
.second-tab p {
  font-size: 2rem;
  text-align: center;
}
