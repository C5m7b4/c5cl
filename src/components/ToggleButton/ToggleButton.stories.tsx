import React from 'react';
import { Story, Meta } from '@storybook/react';

import ToggleButton from './ToggleButton';

export default {
  title: 'Forms/Toggle Button',
  component: ToggleButton,
} as Meta;

const children = [
  <div title="tab1">
    <p></p>
  </div>,
  <div title="tab2">
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatem rem
      adipisci a. Deserunt consequatur labore amet quia modi? Ab explicabo
      cumque tempora sed quo ad quaerat vitae tenetur ipsam blanditiis
      necessitatibus numquam, dolor minus, obcaecati et nulla! Fugiat dolore
      laudantium sunt, optio reprehenderit autem explicabo voluptas officiis
      doloribus ut maiores et veniam consequatur eveniet natus saepe odio
      dolorem cum quidem ullam, dicta, necessitatibus molestiae. Eum, itaque
      quibusdam. Illo perspiciatis libero voluptates error mollitia ab ut
      consequatur iste autem! Ipsam architecto tempore deserunt? Impedit
      laboriosam deleniti nesciunt illo. Blanditiis suscipit autem neque, quod
      impedit praesentium at, excepturi repudiandae totam dolorum temporibus!
    </p>
  </div>,
  <div title="tab3">
    <p>Here is tab 3</p>
  </div>,
];

const Template: Story = (args) => <ToggleButton>{children}</ToggleButton>;

export const ToggleButtonExample = Template.bind({});
ToggleButtonExample.args = {
  children,
};
