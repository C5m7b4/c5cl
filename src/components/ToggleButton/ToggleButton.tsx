import React, { useState, useEffect } from 'react';
import TabNavItem from './TabNavItem';
import TabContent from './TabContent';

import './ToggleButton.css';

export interface ToggleButtonProps {
  children: JSX.Element[];
}

const ToggleButton = (props: ToggleButtonProps) => {
  const [activeTab, setActiveTab] = useState('tab1');
  const { children } = props;

  useEffect(() => {
    setActiveTab('tab-0');
  }, []);

  return (
    <div className="toggle-button">
      <ul className="nav">
        {props.children &&
          props.children.map((child, idx) => (
            <TabNavItem
              key={`tni-${idx}`}
              title={child.props.title}
              id={`tab-${idx}`}
              activeTab={activeTab}
              setActiveTab={setActiveTab}
            />
          ))}
      </ul>
      <div className="outlet">
        {children &&
          children.map((child, idx) => (
            <TabContent
              key={`tc-${idx}`}
              id={`tab-${idx}`}
              activeTab={activeTab}
            >
              {child.props.children}
            </TabContent>
          ))}
      </div>
    </div>
  );
};

export default ToggleButton;
