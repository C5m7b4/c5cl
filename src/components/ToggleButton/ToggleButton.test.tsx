import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ToggleButton from './ToggleButton';

describe('ToggleButton', () => {
  test('should render correctly', () => {
    const { container } = render(
      <ToggleButton>
        <div title="Tab1">
          <p>Here is some content</p>
        </div>
        <div title="tab2">
          <p>Here is some more content</p>
        </div>
      </ToggleButton>
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle click', () => {
    const { container } = render(
      <ToggleButton>
        <div title="tab1">
          <p>Here is some content</p>
        </div>
        <div title="tab2">
          <p>Here is some content</p>
        </div>
      </ToggleButton>
    );
    const div = container.querySelector(
      '.toggle-button-active-tab'
    ) as HTMLDivElement;
    fireEvent.click(div);
  });
});
