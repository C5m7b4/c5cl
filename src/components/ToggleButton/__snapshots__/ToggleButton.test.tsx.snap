// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton should render correctly 1`] = `
<div>
  <div
    class="toggle-button"
  >
    <ul
      class="nav"
    >
      <li
        class="toggle-button-active-tab"
      >
        Tab1
      </li>
      <li
        class=""
      >
        tab2
      </li>
    </ul>
    <div
      class="outlet"
    >
      <div
        class="tab-content"
      >
        <p>
          Here is some content
        </p>
      </div>
    </div>
  </div>
</div>
`;
